src/backend/distributed/executor/adaptive_executor.c

```diff
@@ -2507,6 +2488,28 @@ ConnectionStateMachine(WorkerSession *session)
 }
 
 
+/*
+ * HandleMultiConnectionSuccess logs the established connection and updates connection's state.
+ */
+static void
+HandleMultiConnectionSuccess(WorkerPool *workerPool, MultiConnection *connection,
+							 WorkerSession *session)
+{
+	ereport(DEBUG4, (errmsg("established connection to %s:%d for "
+							"session %ld",
+							connection->hostname, connection->port,
+							session->sessionId)));
+
+	workerPool->activeConnectionCount++;
+	workerPool->idleConnectionCount++;
+
+	UpdateConnectionWaitFlags(session,
+							  WL_SOCKET_READABLE | WL_SOCKET_WRITEABLE);
+
+	connection->connectionState = MULTI_CONNECTION_CONNECTED;
```

Comment:
I'd have a strong preference for keeping the last line in the state machine to clearly show the state change. I'd have a slight preference for keeping these last 2 lines in the state machine, mostly to keep the UpdateConnectionWaitFlags calls together.

Reply:
That makes sense, I will keep the last 2 lines in the state machine.